perf/benchmark/runner/runner.py

```diff
@@ -119,37 +120,48 @@ def __init__(
             sys.exit("invalid mesh %s, must be istio or linkerd" % mesh)
 
     def get_protocol_uri_fragment(self):
-        return "https" if self.mode == "grpc" else "http" 
+        return "https" if self.mode == "grpc" else "http"
+
+    def compute_uri(self, svc, port_type):
+        return "{protocol}://{svc}:{port}/".format(
+            svc=svc, port=self.ports[self.mode][port_type], protocol=self.get_protocol_uri_fragment())
 
     def nosidecar(self):
-        basestr = "{protocol}://{svc}:{port}/"
-        return "base", basestr.format(
-            svc=self.server.ip, port=self.ports[self.mode]["direct_port"], protocol=self.get_protocol_uri_fragment())
+        return "base", self.compute_uri(self.server.ip, "direct_port")
 
     def serversidecar(self):
-        basestr = "{protocol}://{svc}:{port}/"
-        return "serveronly", basestr.format(
-            svc=self.server.ip, port=self.ports[self.mode]["port"], protocol=self.get_protocol_uri_fragment())
+        return "serveronly", self.compute_uri(self.server.ip, "port")
 
-    def clientsidecar(self, nighthawk_cmd):
-        basestr = "{protocol}://{svc}:{port}/"
-        return "base", basestr.format(
-            svc=self.server.ip, port=self.ports[self.mode]["direct_port"], protocol=self.get_protocol_uri_fragment())
+    def clientsidecar(self):
+        return "clientonly", self.compute_uri(self.server.ip, "direct_port")
 
     def bothsidecar(self):
-        basestr = "{protocol}://{svc}:{port}/"
-        return "both", basestr.format(
-            svc=self.server.labels["app"], port=self.ports[self.mode]["port"], protocol=self.get_protocol_uri_fragment())
+        return "both", self.compute_uri(self.server.labels["app"], "port")
 
-    def ingress(self, nighthawk_cmd):
+    def ingress(self):
         url = urlparse(self.run_ingress)
         # If scheme is not defined fallback to http
         if url.scheme == "":
-            url = urlparse("{protocol}://{svc}".format(svc=self.run_ingress), protocol=self.get_protocol_uri_fragment())
-
+            url = urlparse(
+                "{protocol}://{svc}".format(protocol=self.get_protocol_uri_fragment(), svc=self.run_ingress))
         return "ingress", "{url}/".format(url=url.geturl())
 
-    def run(self, conn, qps, size, duration):
+    def execute_mode(self, mode_description, nighthawk_cmd, fn, labels, perf_label_postfix):
+        print('-------------- Running in {mode_description} --------------'.format(
```

Comment:
To make this much interpretable, please name this func to be execute_sidecar_mode, since we have different modes in test configurations: such as `sidecar_mode` and `telemetry_mode`

Reply:
I renamed this, let me know if this looks better.

Reply:
yes, thanks